vb.go

```diff
@@ -0,0 +1,54 @@
+package sgbucket
+
+import (
+	"fmt"
+	"sync"
+)
+
+type VbucketNo uint32
+type VbucketSeq uint64
```

Comment:
I've seen ID types created to alias primitive types before. e.g. `WidgetID int` and `UserID int`. The most useful advantage I've found with that is that it provides type safety when passing arguments to a function.

For example, it's easy to accidentally transpose arguments for this function:

```go
func AssignWidget(widgetID, userID int)
```

Whereas transposing functions for this function would cause a compile-time error:

```go
func AssignWidget(widgetID WidgetID, userID UserID)
```

That being said, some basic unit tests will usually accomplish the same thing. I don't typically alias my primitive types simply because I find the code to be more verbose and harder to read with all the type conversion.

As for the advantages you listed, I don't find that I typically attach methods onto primitive types themselves very often. Also, converting from a primitive type to a `struct` can cause bugs since you usually pass primitives by value and many times `struct`s by reference. For example, if you convert to a struct and add a `sync.Mutex` then copying by value is problematic. However, I don't find that I need to convert primitive types to structs very often.